gammapy/image/maps.py

```diff
@@ -313,12 +331,16 @@ def paste(self, skymap, method='sum'):
         xlo, xhi, ylo, yhi = self._get_boundaries(self, skymap)
         xlo_c, xhi_c, ylo_c, yhi_c = self._get_boundaries(skymap, self)
 
-        if method == 'sum':
-            self.data[ylo:yhi, xlo:xhi] += skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
-        elif method == 'replace':
-            self.data[ylo:yhi, xlo:xhi] = skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
-        else:
-            raise ValueError('Invalid method: {}'.format(method))
+        try:
+            if method == 'sum':
+                self.data[ylo:yhi, xlo:xhi] += skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+            elif method == 'replace':
+                self.data[ylo:yhi, xlo:xhi] = skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+            else:
+                raise ValueError('Invalid method: {}'.format(method))
+        except ValueError:
```

Comment:
This try-except logic doesn't look right?
If the user passes `method='spam'` you raise a ValueError, then catch it, and raise a WCSError?